notifier/sendloop_test.go

```diff
@@ -0,0 +1,205 @@
+	alert4 := &Alert{Labels: labels.FromStrings("alertname", "alert4")}
+	alert5 := &Alert{Labels: labels.FromStrings("alertname", "alert5")}
+	alert6 := &Alert{Labels: labels.FromStrings("alertname", "alert6")}
+
+	// Add new alerts to queue, expect 3 dropped: 1 from new batch + 2 from existing queued items
+	s.add(alert3, alert4, alert5, alert6)
+	require.Equal(t, 3.0, prom_testutil.ToFloat64(s.metrics.dropped.WithLabelValues(s.alertmanagerURL)))
+
+	// Verify all new alerts were added to the queue
+	require.Equal(t, []*Alert{alert4, alert5, alert6}, s.queue)
+}
+
+func TestNextBatchAlertsFromQueue(t *testing.T) {
+	s := newSendLoop("http://foo.bar/", nil, nil, &Options{QueueCapacity: 5, MaxBatchSize: 3}, slog.New(slog.DiscardHandler), newAlertMetrics(prometheus.NewRegistry(), nil))
+
+	alert1 := &Alert{Labels: labels.FromStrings("alertname", "alert1")}
+	alert2 := &Alert{Labels: labels.FromStrings("alertname", "alert2")}
+	alert3 := &Alert{Labels: labels.FromStrings("alertname", "alert3")}
+	s.add(alert1, alert2, alert3)
+
+	// Test batch-size alerts in the queue
+	require.Equal(t, []*Alert{alert1, alert2, alert3}, s.nextBatch())
+	require.Empty(t, s.nextBatch())
+
+	// Test full queue
+	alert4 := &Alert{Labels: labels.FromStrings("alertname", "alert4")}
+	alert5 := &Alert{Labels: labels.FromStrings("alertname", "alert5")}
+	s.add(alert1, alert2, alert3, alert4, alert5)
+	require.Equal(t, []*Alert{alert1, alert2, alert3}, s.nextBatch())
+	require.Equal(t, []*Alert{alert4, alert5}, s.nextBatch())
+	require.Empty(t, s.nextBatch())
+}
+
+func TestMetrics(t *testing.T) {
+	const alertmanagerURL = "http://alertmanager:9093"
+
+	// Use a single registry throughout the test - this is critical to catch registry conflicts
+	reg := prometheus.NewRegistry()
+	alertmanagersDiscoveredFunc := func() float64 { return 0 }
+	metrics := newAlertMetrics(reg, alertmanagersDiscoveredFunc)
+
+	logger := slog.New(slog.DiscardHandler)
+	opts := &Options{QueueCapacity: 10, MaxBatchSize: DefaultMaxBatchSize}
+
+	// Create first sendLoop - this initializes metrics with the alertmanager URL label
+	sendLoop1 := newSendLoop(alertmanagerURL, nil, &config.DefaultAlertmanagerConfig, opts, logger, metrics)
+
+	// Verify metrics are initialized
+	require.Equal(t, 0.0, prom_testutil.ToFloat64(metrics.dropped.WithLabelValues(alertmanagerURL)))
+	require.Equal(t, 0.0, prom_testutil.ToFloat64(metrics.sent.WithLabelValues(alertmanagerURL)))
+	require.Equal(t, 10.0, prom_testutil.ToFloat64(metrics.queueCapacity.WithLabelValues(alertmanagerURL)))
+
+	// Stop the sendLoop - this should clean up all metrics
+	sendLoop1.stop()
+
+	// Verify that the metric labels have been deleted from the registry
+	// We check this by collecting metrics and ensuring our label combination is not present
+	metricFamilies, err := reg.Gather()
+	require.NoError(t, err)
+
+	// Check that none of the metric families contain our alertmanager URL label
+	for _, mf := range metricFamilies {
```

Comment:
thanks for the test.
but some the checks are going way deeper than what I imagined, making the test difficult to read.
I though that calling `newSendLoop` twice with a missing `DeleteLabelValues` would panic, which isn't the case as we are using (counter/gauge)vectors apparently.

let's do without the test for now, we can think about a simpler implementation latter.